build.gradle

```diff
@@ -49,9 +54,9 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.2.1'
+        classpath "com.android.tools.build:gradle:$build_gradle_version"
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath 'com.dicedmelon.gradle:jacoco-android:0.1.2'
+        classpath "com.dicedmelon.gradle:jacoco-android:$jacoco_android_plugin_version"
```

Comment:
Why do we need this? It seems like it's still working without 😊 

Any explanation, please?

cc @sleepylee

Reply:
Yes, we no need to use this plugin for this case so should we remove it @Tuubz @sleepylee

Reply:
Still curious tho, perhaps I missed something? 🤔 

https://github.com/arturdm/jacoco-android-gradle-plugin

cc @sleepylee

Reply:
it has this plugin init: https://github.com/arturdm/jacoco-android-gradle-plugin/blob/master/src/main/resources/META-INF/gradle-plugins/jacoco-android.properties that points to the gradle Task that help generating the Report. You can just add this 1 line to our jacoco properties file and that should be equivalent to this dependency I think 😄

Reply:
Aah, I see. So what this plugins does, is enabling to create reports for multiple variants/flavors automatically? Meaning without we would need to specify it manually?

Reply:
@Tuubz I believe so. This is from the repository README

![image](https://user-images.githubusercontent.com/13435717/78744128-fa4cb400-798a-11ea-9115-fdd5f5534a79.png)

Also, since it's a plugin, it requires 
`apply plugin: 'jacoco-android'`
https://github.com/arturdm/jacoco-android-gradle-plugin

For us, we are creating our own task and only apply the origin jacoco plugin: https://github.com/nimblehq/android-templates/blob/kotlin/config/jacoco.gradle
I'm not sure where did @Wadeewee found this piece of code, but I think not many people used it because of different Kotlin config ever since

Reply:
or maybe they have fixed it already. Then you will just need to use the plugin to reduce the amount of customized Task we are creating here. Our job is to try them out, test if they work, and propose a better solution if it is :)

Reply:
https://github.com/nimblehq/android-templates/blob/82ff99dc3603aae145f3614785507f5248855ac8/app/build.gradle#L3

Reply:
From my understanding is that by specifying the custom tasks ourselves it allows us to have more control over it (Ex: Choosing what files to exclude)

So I don't see the point of specifying both the tasks ourselves + using the plug-in at the same time :/

Reply:
I understand what it does, but rather asking why we chose to integrate this plugin from the start? Even though we had customized tasks specified already. Just trying to get a better understanding on this 😇 🙏

Reply:
no we didn't have this plugin before, @Tuubz 🤔 isn't that your question why @Wadeewee brought this dependency in without using yeah?

Reply:
The plugin was already there before she started on it 🤔

Reply:
ohhh yeah, sorry my bad reading, she only extracts the version out

Reply:
https://github.com/nimblehq/android-templates/pull/16/files#diff-c197962302397baf3a4cc36463dce5eaR52 @Tuubz it was indeed added by me, so I think it was my bad at that time I didn't check it properly after following a few instructions on jacoco setup 🙏 we could potentially just ditch it.

Reply:
Thank you @sleepylee 

@Wadeewee 

Please double check if everything is working as expected without the plugin, if so please remove it 😊

Reply:
I'm sorry @Tuubz @sleepylee I just understand and try that plugin work. it seems like it cannot support multi-module I think, see the log
![Screen Shot 2563-04-08 at 11 57 54](https://user-images.githubusercontent.com/28002315/78746654-a42f3f00-7991-11ea-91cf-dfdfea19991c.png)

Reply:
@jean-t86 @luongvo @Tuubz @sleepylee PR Update.Please review again :)

Reply:
I'm going update POW.